dune/grid/CpGrid.hpp

```diff
@@ -565,23 +567,28 @@ namespace Dune
         // loadbalance is not part of the grid interface therefore we skip it.
 
         /// \brief Distributes this grid over the available nodes in a distributed machine
+        /// \param ecl Pointer to the eclipse state information. Default: null
```

Comment:
Although I know from discussions and the PR intro what this argument is used for, it is not clear in isolation. Could you add a sentence stating that if an eclipse stat is given, well information is extracted and used?